### ISSUE TITLE
Another command already exists for plugin "python"

### ISSUE THREAD

**Issue by kawaiier** [weechat/weechat#1663]
## Bug summary

Another command "invite" already exists for plugin "python"
Another command "kick" already exists for plugin "python"

## Steps to reproduce

1. Installed and set up Matrix account and chats
2. Enabled script download
3. Downloaded Jabber script via `/script load jabber.py`
4. Matrix keeps disconnecting and Jabber won't work at all

## Current behavior
`
19:26:10 matrix_org  -- | matrix: Connecting to matrix.org:443 (SSL)...
26:11    weechat =!= | Another command "invite" already exists for plugin "python"
26:11    weechat =!= | Another command "kick" already exists for plugin "python"
26:11    weechat     | python scripts loaded: jabber, matrix
26:11    weechat     | script: 366 scripts for WeeChat 3.2
26:11    weechat     | Plugins loaded: alias, buflist, charset, exec, fifo, fset, irc, logger, lua, perl, python, relay, ruby, script, spell, tcl, trigger, xfer
26:11 matrix_org  -- | matrix: Doing SSL handshake...
26:11 matrix_org  -- | Error while doing SSL handshake: Unknown error
26:11 matrix_org  -- | matrix: disconnecting from server...
26:11 matrix_org  -- | matrix: disconnected from server
26:11 matrix_org  -- | matrix: reconnecting to server in 10 seconds
`

## Expected behavior
I expect XMPP and Matrix work along at the same time

## Suggested solutions
None

## Additional information

---

- WeeChat version: 3.2
- OS, distribution and version: MacOS 11.4
- Terminal: iTerm 2 Build 3.4.8
- Terminal multiplexer (screen/tmux/…/none):

**Comment by poljar** [comment]
The disconnect issue is a Python bug: https://bugs.python.org/issue44050

As for the commands, that's a Weechat limitation, scripts share the scope of the underlying plugin for many things, commands amongst others. The Matrix script is being rewritten as a plugin because of this and a couple more limitations which plugins inherently don't have.

**Comment by trygveaa** [comment]
> As for the commands, that's a Weechat limitation, scripts share the scope of the underlying plugin for many things, commands amongst others. The Matrix script is being rewritten as a plugin because of this and a couple more limitations which plugins inherently don't have.

You can work around this by using `hook_command_run` instead of `hook_command`. wee-slack does this: https://github.com/wee-slack/wee-slack/blob/v2.8.0/wee_slack.py#L6184

**Comment by weechatter** [comment]
and its also recommended to use "bitlbee" instead of the jabber.py script.

**Comment by flashcode** [comment]
Yes, the conflict of commands inside a same plugin is a known problem.

The script plugins (python and others) already sets a "subplugin" string in the hooks, with the name of the script. But this is used only for the `/help` command and some other things, not for looking for commands when they are executed.

I wonder if it would be easy to make changes to handle this subplugin when we look for a command. One way to do it would be to use the buffer where the command is executed. So if it's executed on a matrix buffer, we know the matrix script created the buffer and then we use first commands with subplugin "matrix". This does not solve the problem when the command is executed in another buffer though, for example the weechat "core" buffer.
